lib/spack/spack/package.py

```diff
@@ -1521,42 +1540,42 @@ def _hms(seconds):
         parts.append("%.2fs" % s)
     return ' '.join(parts)
 
-class StagedPackage(Package):
-    """A Package subclass where the install() is split up into stages."""
-
-    def install_setup(self):
-        """Creates an spack_setup.py script to configure the package later if we like."""
-        raise InstallError("Package %s provides no install_setup() method!" % self.name)
-
-    def install_configure(self):
-        """Runs the configure process."""
-        raise InstallError("Package %s provides no install_configure() method!" % self.name)
-
-    def install_build(self):
-        """Runs the build process."""
-        raise InstallError("Package %s provides no install_build() method!" % self.name)
-
-    def install_install(self):
-        """Runs the install process."""
-        raise InstallError("Package %s provides no install_install() method!" % self.name)
-
-    def install(self, spec, prefix):
-        if 'setup' in self.install_phases:
-            self.install_setup()
-
-        if 'configure' in self.install_phases:
-            self.install_configure()
-
-        if 'build' in self.install_phases:
-            self.install_build()
-
-        if 'install' in self.install_phases:
-            self.install_install()
-        else:
-            # Create a dummy file so the build doesn't fail.
-            # That way, the module file will also be created.
-            with open(os.path.join(prefix, 'dummy'), 'w') as fout:
-                pass
+#class StagedPackage(Package):
+#    """A Package subclass where the install() is split up into stages."""
+#    _phases = ['configure']
+#    def install_setup(self):
+#        """Creates an spack_setup.py script to configure the package later if we like."""
+#        raise InstallError("Package %s provides no install_setup() method!" % self.name)
+#
+#    def install_configure(self):
+#        """Runs the configure process."""
+#        raise InstallError("Package %s provides no install_configure() method!" % self.name)
+#
+#    def install_build(self):
+#        """Runs the build process."""
+#        raise InstallError("Package %s provides no install_build() method!" % self.name)
+#
+#    def install_install(self):
+#        """Runs the install process."""
+#        raise InstallError("Package %s provides no install_install() method!" % self.name)
+#
+#    def install(self, spec, prefix):
+#        if 'setup' in self._phases:
+#            self.install_setup()
+#
+#        if 'configure' in self._phases:
+#            self.install_configure()
+#
+#        if 'build' in self._phases:
+#            self.install_build()
+#
+#        if 'install' in self._phases:
+#            self.install_install()
+#        else:
+#            # Create a dummy file so the build doesn't fail.
+#            # That way, the module file will also be created.
+#            with open(os.path.join(prefix, 'dummy'), 'w') as fout:
+#                pass
```

Comment:
This won't be needed anymore, as the logic in `Package` is generic enough to deal with `CMakePackage` directly